geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/sortedset/AbstractZIncrByIntegrationTest.java

```diff
@@ -0,0 +1,290 @@
+  @Test
+  public void shouldSetScoreToInfinity_ifIncrementWouldExceedMaxValue() {
+    double increment = Double.MAX_VALUE / 2;
+
+    jedis.zadd(STRING_KEY, Double.MAX_VALUE, STRING_MEMBER);
+    jedis.zincrby(STRING_KEY, increment, STRING_MEMBER);
+    assertThat(jedis.zscore(STRING_KEY, STRING_MEMBER)).isEqualTo(POSITIVE_INFINITY);
+
+    jedis.zadd(STRING_KEY, -Double.MAX_VALUE, STRING_MEMBER);
+    jedis.zincrby(STRING_KEY, -increment, STRING_MEMBER);
+    assertThat(jedis.zscore(STRING_KEY, STRING_MEMBER)).isEqualTo(NEGATIVE_INFINITY);
+  }
+
+  @Test
+  public void shouldSetScoreToInfinity_givenInfiniteIncrement() {
+    jedis.zadd(STRING_KEY, 1.0, STRING_MEMBER);
+    assertThat(jedis.zincrby(STRING_KEY, POSITIVE_INFINITY, STRING_MEMBER))
+        .isEqualTo(POSITIVE_INFINITY);
+
+    jedis.zadd(STRING_KEY, 1.0, STRING_MEMBER);
+    assertThat(jedis.zincrby(STRING_KEY, NEGATIVE_INFINITY, STRING_MEMBER))
+        .isEqualTo(NEGATIVE_INFINITY);
+
+    jedis.zadd(STRING_KEY, 1.0, STRING_MEMBER);
+    jedis.sendCommand(STRING_KEY, Protocol.Command.ZINCRBY, STRING_KEY, "inf", STRING_MEMBER);
+    assertThat(jedis.zscore(STRING_KEY, STRING_MEMBER)).isEqualTo(POSITIVE_INFINITY);
+
+    jedis.zadd(STRING_KEY, 1.0, STRING_MEMBER);
+    jedis.sendCommand(STRING_KEY, Protocol.Command.ZINCRBY, STRING_KEY, "+inf", STRING_MEMBER);
+    assertThat(jedis.zscore(STRING_KEY, STRING_MEMBER)).isEqualTo(POSITIVE_INFINITY);
+
+    jedis.zadd(STRING_KEY, 1.0, STRING_MEMBER);
+    jedis.sendCommand(STRING_KEY, Protocol.Command.ZINCRBY, STRING_KEY, "-inf", STRING_MEMBER);
+    assertThat(jedis.zscore(STRING_KEY, STRING_MEMBER)).isEqualTo(NEGATIVE_INFINITY);
+
+    jedis.zadd(STRING_KEY, 1.0, STRING_MEMBER);
+    jedis.sendCommand(STRING_KEY, Protocol.Command.ZINCRBY, STRING_KEY, "Infinity", STRING_MEMBER);
+    assertThat(jedis.zscore(STRING_KEY, STRING_MEMBER)).isEqualTo(POSITIVE_INFINITY);
+
+    jedis.zadd(STRING_KEY, 1.0, STRING_MEMBER);
+    jedis.sendCommand(STRING_KEY, Protocol.Command.ZINCRBY, STRING_KEY, "+Infinity", STRING_MEMBER);
+    assertThat(jedis.zscore(STRING_KEY, STRING_MEMBER)).isEqualTo(POSITIVE_INFINITY);
+
+    jedis.zadd(STRING_KEY, 1.0, STRING_MEMBER);
+    jedis.sendCommand(STRING_KEY, Protocol.Command.ZINCRBY, STRING_KEY, "-Infinity", STRING_MEMBER);
+    assertThat(jedis.zscore(STRING_KEY, STRING_MEMBER)).isEqualTo(NEGATIVE_INFINITY);
+  }
+
+  @Test
+  public void nullRedisSetShouldCreateNonExistentMember_withIncrementOfPositiveInfinity() {
+    final double increment = POSITIVE_INFINITY;
+    jedis.zadd(KEY, 0.0, "something".getBytes()); // init the key but not the member
+
+    jedis.zincrby(KEY, increment, MEMBER);
+    assertThat(jedis.zscore(KEY, MEMBER)).isEqualTo(increment);
+  }
+
+  @Test
+  public void nullRedisSetShouldCreateNonExistentMember_withIncrementOfNegativeInfinity() {
+    final double increment = NEGATIVE_INFINITY;
+    jedis.zadd(KEY, 0.0, "something".getBytes()); // init the key but not the member
```

Comment:
Another call to `getBytes()` that could be replaced with `Coder.stringToBytes()`.